### ISSUE TITLE
WeaponCollection.Give Exception

### ISSUE THREAD

**Issue by Dilapidated** [crosire/scripthookvdotnet#428]
So I've been using the beta DotNetScriptHook from here https://ci.appveyor.com/project/crosire/scripthookvdotnet due to issues with WeaponCollection.Give following the 2.5.1 release. It resolved this issue until recently and now appears to be back in the beta release. I can't say for sure which build the issue returned in but I know it wasn't an issue in the beta build I grabbed around the 15th of last month.

Basically the WeaponCollection.Give causes exceptions when giving the same weapon twice and can cause CTD issues for some users. More specifically it's been causing a lot of users errors in the script me and my team are working on: https://www.gta5-mods.com/misc/realism-dispatch-enhanced

Could you kindly check to see why this code isn't working in the beta version anymore and re-add the fix?

**Comment by kagikn** [comment]
I couldn't confirmed there is a bug in WeaponCollection.Give, I tested it with following code and this code works fine for me (on Build No. 681).
```
                var player = Game.Player.Character;
                if (player == null || !player.Exists()) { return; }
                player.Weapons.Give(WeaponHash.APPistol, 100, true, true);
                player.Weapons.Give(WeaponHash.APPistol, 100, true, true);
```

**Comment by Dilapidated** [comment]
Thanks for investigating. It's strange that the beta was working correctly until recently. I'll be home in a few hours and will turn on debug to gather more info for you. Is there anyway to force debugging a .dll to show line numbers like the .cs does? (can't use .cs due to it not allowing references like .xml etc.)

**Comment by kagikn** [comment]
Noose Riots spawn fine for me, too. But I'm not sure how to force debugging a .dll, sorry.

**Comment by Dilapidated** [comment]
I didn't have issues either during my tests either. I did some more research and verified with those who were having issues.... it turns out that they had corrupted the install of the juggernaut model which lead to the CTD and invisible Noose Riots.

Issue is resolved.

Thank you again for looking into this with me :)

**Comment by Dilapidated** [comment]
I didn't have issues either during my tests either. I did some more research and verified with those who were having issues.... it turns out that they had corrupted the install of the juggernaut model which lead to the CTD and invisible Noose Riots.

Issue is resolved.

Thank you again for looking into this with me :)
